### PR TITLE
Convert to use inverted build tags for pkcs11

### PR DESCRIPTION
**Pull request by runcom**
ping @jfrazelle 
Fix https://github.com/docker/docker/pull/17993

Signed-off-by: Antonio Murdaca <runcom@redhat.com>

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -86,7 +85,8 @@ fmt:
 
 lint:
 	@echo "+ $@"
-	@test -z "$$(golint ./... | grep -v .pb. | grep -v Godeps/_workspace/src/ | tee /dev/stderr)"
+	# TODO: remove trustmanager/yubikey/ from ignored once linted
```
Comment (runcom) [inline Makefile]: just didn't want to overload this PR (plus I'm not so confident in documenting that right now)

**Comment by runcom** [comment]
This way we should add `exclude_pkcs11` in stable docker build and nothing in experimental

**Comment by endophage** [comment]
Just curious, is there a docker/docker preference for excluding features rather than including? In the past I've generally preferred that the minimal command (i.e. no build tags) should build/run stable.

File: Makefile (hunk 1 above)
Comment (jfrazelle) [inline Makefile]: No no can you do in this PR I added this as a safety

Comment (jfrazelle) [inline Makefile]: Oh nm well idk I have a bad feeling it's going to cause something bad eventually

Comment (runcom) [inline Makefile]: alright, I hope I'll get everything right then..

Comment (runcom) [inline Makefile]: btw, before this PR this pkg wasn't linted at all :/

Comment (jfrazelle) [inline Makefile]: My bad I thought this was my test if there was pkcs11 in a non build tag

Comment (jfrazelle) [inline Makefile]: But regarding that... It should probably be updated to include the new tag sorry

Comment (runcom) [inline Makefile]: @jfrazelle  you mean here right? https://github.com/docker/notary/pull/305/files#diff-b67911656ef5d18c4ae36cb6741b7965L23

Comment (jfrazelle) [inline Makefile]: But ya sorry do whatever w the lint hahaha not my project :P

Comment (jfrazelle) [inline Makefile]: It's this https://github.com/runcom/notary/blob/invert-pkcs11-buildtag/Makefile#L21

Comment (runcom) [inline Makefile]: yeah I meant that but link doesn't point there :/, I'll update that to be `!exclude_pkcs11`

Comment (jfrazelle) [inline Makefile]: Awesome thanks and sorry for the confusion

Comment (runcom) [inline Makefile]: Updated as per your comment @jfrazelle

**Comment by runcom** [comment]
@endophage this is because pkcs11 is coming with experimental for now, usually it should be like you said I think

**Comment by endophage** [comment]
I'm not following. I'd expect experimental to be stable + enabling build tags.

**Comment by runcom** [comment]
> Just curious, is there a docker/docker preference for excluding features rather than including? In the past I've generally preferred that the minimal command (i.e. no build tags) should build/run stable.

The minimal command will continue to build/run stable as soon as pkcs11 will get in stable.

Right now the issue is that there's no way with Docker+experimental to opt-out pkcs11 w/o breaking builds (https://github.com/docker/docker/pull/17993#issuecomment-156799314)
Experimental will be stable+enabled **experimental** build tag and for now we well exclude pkcs11 from stable (@jfrazelle can you confirm I haven't totally misunderstood?)
To the first question docker/docker has `exclude_*` tags used in the graphdrivers for instance (i.e. opt-out overlay or aufs), I don't know if it's a preference tho.
Does all of this feel wrong btw?

**Comment by endophage** [comment]
Based on the docker/docker PR you've linked, it seems like you're already providing quite the litany of build flags so rather than setting the `DOCKER_EXPERIMENTAL` env var, would it not simply make more sense to just directly provide the `experimental` build tag, which would also give you the flexibility to not provide the pkcs11 tag?

**Comment by runcom** [comment]
@endophage you're right, setting just `experimental` works fine and solves my issue...thanks, I'll close both

**Pull request closed without merging**